src/dmd/func.d

```diff
@@ -3088,6 +3088,20 @@ private bool checkEscapingSiblings(FuncDeclaration f, FuncDeclaration outerFunc,
             bAnyClosures = true;
         }
 
+        FuncDeclaration parent = g;
+        while (!bAnyClosures)
+        {
+            parent = parent.parent.isFuncDeclaration;
+            if (!parent || parent is outerFunc) break;
+            // A parent of the sibling had its address taken.
+            // Assume escaping of parent affects its children, so needs propagating.
```

Comment:
please add a bugzilla reference to this comment

Reply:
added